docs/user_guide/mitigation.rst

```diff
@@ -80,11 +81,408 @@ Reductions
+    >>> y_true             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  0 ,  0 ,  0 ])
+    >>> y_pred             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  0 ,  0 ,  0 ])
+    >>> sensitive_features = np.array(["a", "b", "a", "a", "b", "a", "b", "b", "a", "b"])
+    >>> selection_rate_group_summary(y_true, y_pred, sensitive_features=sensitive_features)
+    {'overall': 0.4, 'by_group': {'a': 0.6, 'b': 0.2}}
+    >>> dp.load_data(X, y_true, sensitive_features=sensitive_features)
+    >>> dp.gamma(lambda X: y_pred)
+    sign  event  group_id
+    +     all    a           0.2
+                 b          -0.2
+    -     all    a          -0.2
+                 b           0.2
+    dtype: float64
+ 
+Rewriting this into a ratio-based constraint with :code:`ratio_bound`
+:math:`r` we get
+
+.. math::
+
+    r \leq \dfrac{\P[h(X) = y' \given A = a]}{\P[h(X) = y']} \leq \dfrac{1}{r} \; \forall a, y'
+
+Revisiting the same example from above we get
+
+.. doctest:: mitigation
+
+    >>> dp = DemographicParity(ratio_bound=0.9, ratio_bound_slack=0.01)
+    >>> dp.load_data(X, y_pred, sensitive_features=sensitive_features)
+    >>> dp.gamma(lambda X: y_pred)
+    sign  event  group_id
+    +     all    a           0.14
+                 b          -0.22
+    -     all    a          -0.24
+                 b           0.16
+    dtype: float64
+
+Just like :code:`difference_bound` before :code:`ratio_bound_slack` does not
+affect the outcome here, because no mitigation is applied in the example.
+This time the constraint violation is calculated using the :code:`ratio_bound`
+`0.9` and again with the definition from
+:ref:`constraints_binary_classification` and selection rate as the used
+metric:
+
+.. math::
+
+    r \cdot \text{metric}_a - \text{metric}_* = 0.9 \times 0.6 - 0.4 = 0.14 \\
+    r \cdot \text{metric}_b - \text{metric}_* = 0.9 \times 0.2 - 0.4 = -0.22 \\
+    - \text{metric}_a + r \cdot \text{metric}_* = - 0.6 + 0.9 \times 0.4 = -0.24 \\
+    - \text{metric}_b + r \cdot \text{metric}_* = - 0.2 + 0.9 \times 0.4 = 0.16 \\
+
+.. _true_positive_rate_parity:
+.. _false_positive_rate_parity:
+
+True Positive Rate Parity and False Positive Rate Parity
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A classifier :math:`h(X)` satisfies *True Positive Rate Parity* if
+
+.. math::
+
+    \P[h(X) = 1 \given A = a, Y = 1] = \P[h(X) = 1 \given Y = 1] \; \forall a
+
+and *False Positive Rate Parity* if
+
+.. math::
+
+    \P[h(X) = 1 \given A = a, Y = 0] = \P[h(X) = 1 \given Y = 0] \; \forall a
+
+Note that this formulation of True Positive Rate Parity ignores what happens
+to samples with :math:`Y=0`, and the False Positive Rate Parity ignores
+:math:`Y=1`.
+Refer to :ref:`equalized_odds` for a constraint that extends this by
+considering all samples, i.e., the samples with :math:`Y=0` as well as the
+ones with :math:`Y=1`.
+
+In practice this can be used in a difference-based version as follows:
```

Comment:
I would start off here by saying that you're going to construct a dataset with given TPR values (overall, a and b). Otherwise people have to pick too much out of the code.

Reply:
You mean write the content of the printed dictionary before the code?
```
{'overall': 0.5714285714285714, 'by_group': {'a': 0.75, 'b': 0.3333333333333333}}
```

Reply:
Effectively - but written as equations and fractions, not just spit out a Python dictionary.